tests/components/velbus/test_services.py

```diff
@@ -0,0 +1,169 @@
+    await init_integration(hass, config_entry)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SCAN,
+        {CONF_INTERFACE: config_entry.data["port"]},
+        blocking=True,
+    )
+    config_entry.runtime_data.controller.scan.assert_called_once_with()
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SYNC,
+        {CONF_INTERFACE: config_entry.data["port"]},
+        blocking=True,
+    )
+    config_entry.runtime_data.controller.sync_clock.assert_called_once_with()
+
+    # Test invalid interface
+    with pytest.raises(vol.error.MultipleInvalid):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SCAN,
+            {CONF_INTERFACE: "nonexistent"},
+            blocking=True,
+        )
+
+    # Test missing interface
+    with pytest.raises(vol.error.MultipleInvalid):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SCAN,
+            {},
+            blocking=True,
+        )
+
+
+async def test_global_survices_with_config_entry(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test services directed at the bus with a config_entry."""
+    await init_integration(hass, config_entry)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SCAN,
+        {CONF_CONFIG_ENTRY: config_entry.entry_id},
+        blocking=True,
+    )
+    config_entry.runtime_data.controller.scan.assert_called_once_with()
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SYNC,
+        {CONF_CONFIG_ENTRY: config_entry.entry_id},
+        blocking=True,
+    )
+    config_entry.runtime_data.controller.sync_clock.assert_called_once_with()
+
+    # Test invalid interface
+    with pytest.raises(ServiceValidationError):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SCAN,
+            {CONF_CONFIG_ENTRY: "nonexistent"},
+            blocking=True,
+        )
+
+    # Test missing interface
+    with pytest.raises(vol.error.MultipleInvalid):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SCAN,
+            {},
+            blocking=True,
+        )
+
+
+async def test_set_memo_text(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    controller: AsyncMock,
+) -> None:
+    """Test the set_memo_text service."""
+    await init_integration(hass, config_entry)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_MEMO_TEXT,
+        {
+            CONF_CONFIG_ENTRY: config_entry.entry_id,
+            CONF_MEMO_TEXT: "Test",
+            CONF_ADDRESS: 1,
+        },
+        blocking=True,
+    )
+    config_entry.runtime_data.controller.get_module(
+        1
+    ).set_memo_text.assert_called_once_with("Test")
+
+    # Test with unfound module
+    controller.return_value.get_module.return_value = None
+    with pytest.raises(ServiceValidationError):
```

Comment:
```suggestion
    with pytest.raises(ServiceValidationError, match=".."):
```
Let's match on the error